trustfall_proc_macros/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use quote::quote;
+
+#[proc_macro_derive(VariantsAsVertexTypes)]
```

Comment:
It would be very nice with docs for _Before_ and _After_ this macro, so it's clear what you're actually adding. Reading the code will be of limited help if you don't know what you're looking for.

(Perhaps `cargo-extend` could make this easy)